Amazon.QLDB.Driver/driver/ISerializer.cs

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver
+{
+    using Amazon.QLDBSession.Model;
+
+    /// <summary>
+    /// Interface for the Ion serializer.
+    /// </summary>
+    public interface ISerializer
+    {
+        /// <summary>
+        /// Serialize a C# object into a ValueHolder object containing the Ion binary or text value.
+        /// </summary>
+        ///
+        /// <param name="o">The C# object to be serialized into ValueHolder.</param>
+        ///
+        /// <returns>The ValueHolder object containing the Ion binary or text value.</returns>
+        ValueHolder Serialize(object o);
```

Comment:
Any reason this is type object instead of type T?

Reply:
I think we should update it, I see in the [Object mapper](https://github.com/amzn/ion-object-mapper-dotnet/blob/main/Amazon.Ion.ObjectMapper/IonSerializer.cs#L170-L176) we are using type `T` as well.

Thoughts @alpian  ?